src/logical/builder_proxy.rs

```diff
@@ -12,10 +12,11 @@ use crate::{
 };
 
 /// Trait capturing builder proxies that use plain string (used for parsing in logical layer)
+///
+/// This parses from a given logical type to the physical type, without exposing details from one layer to the other.
 pub trait LogicalColumnBuilderProxy<'a, 'b>: ColumnBuilderProxy<String> {
-    /// Create a new [`LogicalColumnBuilderProxy`] which nests its physical target equivalent [`BuilderProxy`][crate::physical::builder_proxy::PhysicalBuilderProxyEnum].
+    /// Create a new [`LogicalColumnBuilderProxy`] which is constructed from a given [`BuilderProxy`][crate::physical::builder_proxy::PhysicalBuilderProxyEnum].
```

Comment:
Again, passive voice is not precise enough here – it conveys less information while being harder to read: This _creates a `LogicalColumnBuilderProxy` from_ the given `BuilderProxy`, it's not just creating a `LogicalColumnBuilderProxy` which also (by pure coincidence) happens to somehow be constructed from the given `BuilderProxy`.